Search/Exception/MetadataNotFoundException.php

```diff
@@ -0,0 +1,15 @@
+<?php
+/*
+ * This file is part of the Sulu CMS.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Massive\Bundle\SearchBundle\Search\Exception;
+
+class MetadataNotFoundException extends \RuntimeException
```

Comment:
I would still also pass the class for which no metadata was found to the constructor, so that you can act accordingly to it in the catch block. Just an error message is too less IMHO